usaspending_api/bulk_download/v2/views.py

```diff
@@ -337,42 +337,66 @@ class ListMonthylDownloadsViewset(APIView):
 
     s3_handler = S3Handler(name=settings.MONTHLY_DOWNLOAD_S3_BUCKET_NAME, region=settings.BULK_DOWNLOAD_AWS_REGION)
 
+    @cache_response()
```

Comment:
How long does this request take when it doesn't hit the cache?

Reply:
Depends on the filters. If no filters are provided, it takes ~60s. If the fiscal year is provided, it takes ~5s. If the fiscal year and agency are provided, it takes ~0.5s. This is due to the major bottlenecks being getting the filenames from the s3 bucket (which speeds up faster when a prefix is provided, larger prefixes -> faster results) and getting the agency name/acronym from the results returned as it checks with the database. Those longer cases though will not show up as the front end currently *requires* all three filters so the average will be 0.5s for the endpoint without caching.